### ISSUE TITLE
[Feature Request]: Play next video after removing current video from the playlist

### ISSUE THREAD

**Issue by jasonhenriquez** [FreeTubeApp/FreeTube#4918]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

1. Enable autoplay
2. Open the non-first video of a playlist
3. Remove that video from the playlist while watching it
4. Let video autoplay. Result: first video in the playlist plays

### Proposed Solution

1. Enable autoplay
2. Open the non-first video of a playlist
3. Remove that video from the playlist while watching it
4. Let video autoplay. Result: video that would have been next in the playlist if the current video was not removed from the playlist plays

### Alternatives Considered

N/A

### Issue Labels

ease of use improvement, improvement to existing feature

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I almost consider this a bug but we didnt think about this in testing so i understand why you decided to create a feature request for this

**Comment by jasonhenriquez** [comment]
> I almost consider this a bug but we didnt think about this in testing so i understand why you decided to create a feature request for this

Yeah, I know it can get a bit sensitive/pedantic on what counts as a feature request versus a bug, so I wanted to prevent any potential mini-argument lol.

I am on a bit of a hiatus, but one way I could see this being implemented is by having a `prop` of the last non-`-1` `videoIndexInPlaylistItems` value and using that instead in `watch-video-playlist`'s `playNextVideo()`.

**Comment by jasonhenriquez** [comment]
I've tried thinking about this one tonight and have ended up in extreme overthinking territory. Note to the implementer on this one: please don't overthink, just do the suggestion above. It's impossible to not have things fall apart outside of much more than this one basic case because of the arbitrariness of "maintaining position" with deleted elements. Just do what I suggest above and eff the edge cases, because YT seems to fail so many of them on this specific issue in my testing (and rightfully so, because the non-basic cases are so rare).
